### PR TITLE
(feat) Add config property for rendering the `Create appointment`  button

### PR DESCRIPTION
**Pull request by CynthiaKamau**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
Add config to show/hide create appointment button

## Screenshots
https://www.loom.com/share/929bc5bbebb2442a86c1783639c1d1f5?sid=b9e9b80b-0044-46f5-9741-de5f10237640

*None.*

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Review by mogoodrich — APPROVED** [review]
Yes @CynthiaKamau  this seems good to me, but as @mseaton  says, let's create tickets for all our changes going forward, so we can document in them the reason for a change. Thanks!

File: packages/esm-appointments-app/src/config-schema.ts [hunk 1]
```diff
@@ -71,6 +71,11 @@ export const configSchema = {
     _description: 'The name of the patient identifier type to be used for the patient identifier field',
     _default: 'OpenMRS ID',
   },
+  showCreateAppointmentButtons: {
```
Comment (denniskigen) [inline packages/esm-appointments-app/src/config-schema.ts:74]:
```suggestion
  showCreateAppointmentButton: {
```

File: packages/esm-appointments-app/src/config-schema.ts [hunk 2]
```diff
@@ -71,6 +71,11 @@ export const configSchema = {
...
+    _type: Type.Boolean,
+    _description: 'Whether to show the create appointment button`',
```
Comment (denniskigen) [inline packages/esm-appointments-app/src/config-schema.ts:76]:
```suggestion
    _description: 'Determines whether or not to render the `Create Appointment` button in the appointments page header',
```

File: packages/esm-appointments-app/src/appointments-metrics/metrics-header.component.tsx [hunk 3]
```diff
@@ -32,19 +33,21 @@ const MetricsHeader: React.FC = () => {
...
-            buttonText: t('createNewAppointment', 'Create new appointment'),
-            overlayHeader: t('createNewAppointment', 'Create new appointment'),
-            buttonProps: {
-              kind: 'primary',
-              renderIcon: (props) => <Hospital size={32} {...props} />,
-              size: 'lg',
-            },
-          }}
-        />
+        {showCreateAppointmentButtons && (
```
Comment (denniskigen) [inline packages/esm-appointments-app/src/appointments-metrics/metrics-header.component.tsx:36]:
```suggestion
        {showCreateAppointmentButton && (
```

File: packages/esm-appointments-app/src/appointments-metrics/metrics-header.component.tsx [hunk 4]
```diff
@@ -5,14 +5,15 @@ dayjs.extend(isToday);
...
-import { ExtensionSlot, navigate } from '@openmrs/esm-framework';
+import { ExtensionSlot, navigate, useConfig } from '@openmrs/esm-framework';
 import styles from './metrics-header.scss';
 import { spaBasePath } from '../constants';
 import AppointmentForm from '../appointments/forms/create-edit-form/appointments-form.component';
 import { closeOverlay, launchOverlay } from '../hooks/useOverlay';
 
 const MetricsHeader: React.FC = () => {
   const { t } = useTranslation();
+  const { showCreateAppointmentButtons } = useConfig();
```
Comment (denniskigen) [inline packages/esm-appointments-app/src/appointments-metrics/metrics-header.component.tsx:16]:
```suggestion
  const { showCreateAppointmentButton } = useConfig();
```

File: packages/esm-appointments-app/src/empty-state/empty-state.component.tsx [hunk 5]
```diff
@@ -29,7 +30,7 @@ export const EmptyState: React.FC<EmptyStateProps> = ({ headerTitle, displayText
             There are no {displayText} to display
           </Trans>
         </p>
-        {scheduleType === 'Scheduled' ? (
+        {scheduleType === 'Scheduled' && showCreateAppointmentButtons ? (
```
Comment (denniskigen) [inline packages/esm-appointments-app/src/empty-state/empty-state.component.tsx:33]:
```suggestion
        {scheduleType === 'Scheduled' && showCreateAppointmentButton ? (
```

**Pull request closed without merging**